python/foglamp/services/core/api/notification.py

```diff
@@ -0,0 +1,517 @@
+from foglamp.common.storage_client.payload_builder import PayloadBuilder
+from foglamp.common.storage_client.exceptions import StorageServerError
+from foglamp.common.configuration_manager import ConfigurationManager
+from foglamp.services.core import connect
+from foglamp.services.core.service_registry.service_registry import ServiceRegistry
+from foglamp.services.core.service_registry import exceptions as service_registry_exceptions
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2018 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_help = """
+    -------------------------------------------------------------------------------
+    | GET                            | /foglamp/notification/plugin               |
+    | GET POST PUT DELETE            | /foglamp/notification                      |
+    -------------------------------------------------------------------------------
+"""
+
+_logger = logger.setup()
+
+
+async def get_plugin(request):
+    """ GET lists of rule plugins and delivery plugins
+
+    :Example:
+        curl -X GET http://localhost:8081/foglamp/notification/plugin
+    """
+    try:
+        notification_service = ServiceRegistry.get(s_type=ServiceRecord.Type.Notification.name)
+        _address, _port = notification_service[0]._address, notification_service[0]._management_port
+    except service_registry_exceptions.DoesNotExist:
+        return web.json_response({'result': "No Notification service available."})
+
+    try:
+        url = 'http://{}:{}/foglamp/notification/rules'.format(_address, _port)
+        rule_plugins = json.loads(await _hit_get_url(url))
+
+        url = 'http://{}:{}/foglamp/notification/delivery'.format(_address, _port)
+        delivery_plugins = json.loads(await _hit_get_url(url))
+    except Exception as ex:
+        raise web.HTTPInternalServerError(reason=ex)
+    else:
+        return web.json_response({'rules': rule_plugins, 'delivery': delivery_plugins})
+
+
+async def get_notification(request):
+    """ GET an existing notification
+
+    :Example:
+        curl -X GET http://localhost:8081/foglamp/notification/<notification_name>
+    """
+    try:
+        notif = request.match_info.get('notification_name', None)
+        if notif is None:
+            raise web.HTTPInternalServerError(reason="Notification name is required.")
+
+        notification = {}
+        storage = connect.get_storage_async()
+        config_mgr = ConfigurationManager(storage)
+        notification_config = await config_mgr._read_category_val(notif)
+        if notification_config:
+            rule_config = await config_mgr._read_category_val("rule{}".format(notif))
+            channel_config = await config_mgr._read_category_val("delivery{}".format(notif))
+            notification = {
+                "name": notification_config['name']['value'],
+                "description": notification_config['description']['value'],
```

Comment:
Not aligned as per specs; though `description` field is not available under `get_notifications`

Reply:
added a comment for this in the doc.